<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="test_legs_urdf">
  <!-- <link name="world"/>
    <joint name="base_joint" type="fixed">
      <parent link="world"/>
      <child link="base_link"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.17"/>
    </joint> -->
  <link name="base_link">
    <inertial>
      <origin
        xyz="0.0023973 0.0026111 -0.1"
        rpy="0 0 0" />
      <mass
        value="0.37523" />
      <inertia
        ixx="0.00045474"
        ixy="-3.2434E-08"
        ixz="-7.902E-08"
        iyy="0.00045942"
        iyz="-2.2026E-05"
        izz="0.00084361" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link name="RLeg">
    <inertial>
      <origin
        xyz="-0.010304 0.00087314 0.0095798"
        rpy="0 0 0" />
      <mass
        value="0.013241" />
      <inertia
        ixx="2.8521E-06"
        ixy="-3.5111E-09"
        ixz="-3.048E-08"
        iyy="7.6357E-06"
        iyz="-2.2658E-07"
        izz="7.3792E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RLeg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RLeg.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RHip_yaw"
    type="revolute">
    <origin
      xyz="-0.042665 -0.0164 -0.077419"
      rpy="3.1416 0 1.5708" />
    <parent
      link="base_link" />
    <child
      link="RLeg" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="RHip">
    <inertial>
      <origin
        xyz="0.0091581 -0.011371 0.0048311"
        rpy="0 0 0" />
      <mass
        value="0.049473" />
      <inertia
        ixx="1.107E-05"
        ixy="-1.929E-07"
        ixz="-5.1455E-07"
        iyy="1.7253E-05"
        iyz="-8.0277E-07"
        izz="1.6988E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RHip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RHip.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RHip_roll"
    type="revolute">
    <origin
      xyz="0 0.0055 0.029"
      rpy="0 0 -1.5708" />
    <parent
      link="RLeg" />
    <child
      link="RHip" />
    <axis
      xyz="0 -1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="RFemur">
    <inertial>
      <origin
        xyz="0.0089374 0.00056334 0.018422"
        rpy="0 0 0" />
      <mass
        value="0.053764" />
      <inertia
        ixx="2.1966E-05"
        ixy="-2.7015E-08"
        ixz="-1.8034E-06"
        iyy="2.895E-05"
        iyz="-6.4632E-07"
        izz="1.4647E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RFemur.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RFemur.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RHip_pitch"
    type="revolute">
    <origin
      xyz="0 -0.0066192 0.039"
      rpy="0 0 0" />
    <parent
      link="RHip" />
    <child
      link="RFemur" />
    <axis
      xyz="-1 0 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="RTibia">
    <inertial>
      <origin
        xyz="-0.01514 -0.021484 -0.0081987"
        rpy="0 0 0" />
      <mass
        value="0.081897" />
      <inertia
        ixx="2.4638E-05"
        ixy="-3.6592E-06"
        ixz="1.7308E-09"
        iyy="1.8401E-05"
        iyz="2.2922E-07"
        izz="2.6132E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RTibia.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RTibia.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RKnee_pitch"
    type="revolute">
    <origin
      xyz="0 0.00725 0.0664"
      rpy="-1.5708 0 1.5708" />
    <parent
      link="RFemur" />
    <child
      link="RTibia" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="RAnkle">
    <inertial>
      <origin
        xyz="2.1083E-11 -0.0065876 0.03259"
        rpy="0 0 0" />
      <mass
        value="0.024453" />
      <inertia
        ixx="1.578E-05"
        ixy="1.5525E-14"
        ixz="1.6802E-14"
        iyy="1.4131E-05"
        iyz="-1.4664E-06"
        izz="1.5924E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RAnkle.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RAnkle.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RAnkle_pitch"
    type="revolute">
    <origin
      xyz="-0.0209 -0.04901 0"
      rpy="1.5708 0 0" />
    <parent
      link="RTibia" />
    <child
      link="RAnkle" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="RFoot">
    <inertial>
      <origin
        xyz="0.12479 0.009339 -0.0045296"
        rpy="0 0 0" />
      <mass
        value="0.068473" />
      <inertia
        ixx="2.3765E-05"
        ixy="5.5166E-07"
        ixz="1.0494E-06"
        iyy="4.5323E-05"
        iyz="-6.2769E-07"
        izz="6.1712E-05" />
    </inertial>
    <visual>
      <origin
        xyz="-0.125 0 0.01"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RFoot.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="-0.125 0 0.01"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/RFoot.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="RAnkle_roll"
    type="revolute">
    <origin
      xyz="0.0 0.0051692 0.0626"
      rpy="0 0 3.1416" />
    <parent
      link="RAnkle" />
    <child
      link="RFoot" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LLeg">
    <inertial>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <mass
        value="0.013241" />
      <inertia
        ixx="2.8521E-06"
        ixy="-3.5111E-09"
        ixz="-3.048E-08"
        iyy="7.6357E-06"
        iyz="-2.2658E-07"
        izz="7.3792E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LLeg.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LLeg.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LHip_yaw"
    type="revolute">
    <origin
      xyz="0.048335 -0.0153 -0.077419"
      rpy="3.1416 0 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="LLeg" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LHip">
    <inertial>
      <origin
        xyz="-0.0048311 0.0090773 -0.011371"
        rpy="0 0 0" />
      <mass
        value="0.049473" />
      <inertia
        ixx="1.6988E-05"
        ixy="5.1455E-07"
        ixz="8.0277E-07"
        iyy="1.107E-05"
        iyz="-1.929E-07"
        izz="1.7253E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LHip.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LHip.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LHip_roll"
    type="revolute">
    <origin
      xyz="0 0.0055 0.029"
      rpy="3.1416 1.5708 0" />
    <parent
      link="LLeg" />
    <child
      link="LHip" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LFemur">
    <inertial>
      <origin
        xyz="-0.00033744 -0.018648 -0.0087374"
        rpy="0 0 0" />
      <mass
        value="0.053764" />
      <inertia
        ixx="2.895E-05"
        ixy="-6.4632E-07"
        ixz="-2.7013E-08"
        iyy="1.4647E-05"
        iyz="-1.8034E-06"
        izz="2.1966E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LFemur.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LFemur.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LHip_pitch"
    type="revolute">
    <origin
      xyz="-0.039 0 -0.0066192"
      rpy="1.5708 1.5708 0" />
    <parent
      link="LHip" />
    <child
      link="LFemur" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LTibia">
    <inertial>
      <origin
        xyz="-0.021252 0.015102 0.0079987"
        rpy="0 0 0" />
      <mass
        value="0.081897" />
      <inertia
        ixx="1.8401E-05"
        ixy="3.6592E-06"
        ixz="-2.2922E-07"
        iyy="2.4638E-05"
        iyz="1.7333E-09"
        izz="2.6132E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LTibia.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LTibia.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LKnee_pitch"
    type="revolute">
    <origin
      xyz="-0.00675 -0.0669 0"
      rpy="-3.1416 0 1.5708" />
    <parent
      link="LFemur" />
    <child
      link="LTibia" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LAnkle">
    <inertial>
      <origin
        xyz="-2.1083E-11 -0.0063876 0.03259"
        rpy="0 0 0" />
      <mass
        value="0.024453" />
      <inertia
        ixx="1.578E-05"
        ixy="-1.5525E-14"
        ixz="-1.6802E-14"
        iyy="1.4131E-05"
        iyz="-1.4664E-06"
        izz="1.5924E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LAnkle.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LAnkle.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LAnkle_pitch"
    type="revolute">
    <origin
      xyz="-0.04856 0.02095 0"
      rpy="-1.5708 0 1.5708" />
    <parent
      link="LTibia" />
    <child
      link="LAnkle" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>
  <link name="LFoot">
    <inertial>
      <origin
        xyz="0.00088259 0.0091453 0.0054704"
        rpy="0 0 0" />
      <mass
        value="0.068473" />
      <inertia
        ixx="2.3765E-05"
        ixy="-5.5166E-07"
        ixz="-1.0494E-06"
        iyy="4.5323E-05"
        iyz="-6.277E-07"
        izz="6.1712E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LFoot.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://test_legs_urdf/meshes/LFoot.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="LAnkle_roll"
    type="revolute">
    <origin
      xyz="0 0.0049192 0.062486"
      rpy="0 0 3.1416" />
    <parent
      link="LAnkle" />
    <child
      link="LFoot" />
    <axis
      xyz="-1 0 0" />
    <limit
      lower="-1.57"
      upper="1.57"
      effort="0.3"
      velocity="5" />
  </joint>


  <!-- 
  Adding actuators for every joint
   Replace the “link_n” part with the name of child link of the joint and “joint_n” with the exact joint name for which you are writing this transmission tag. 
   The transmission and actuator names can be defined anything you want. 
   But the joint name must be the exact name of the joint.
  <transmission name="link_n_trans">
  <type>transmission_interface/SimpleTransmission</type>
  <joint name="joint_n">
  <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface></joint>
  <actuator name="link_n_motor">
  <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface> <mechanicalReduction>1</mechanicalReduction>
  </actuator>
  </transmission>
     -->

  <transmission name="LLeg_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LHip_yaw">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LHip_yaw_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="LHip_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LHip_roll">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LHip_roll_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="LFemur_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LHip_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LHip_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="LTibia_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LKnee_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LKnee_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="LAnkle_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LAnkle_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LAnkle_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="LFoot_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="LAnkle_roll">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="LAnkle_roll_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>



  <transmission name="RLeg_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RHip_yaw">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RHip_yaw_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="RHip_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RHip_roll">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RHip_roll_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="RFemur_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RHip_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RHip_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="RTibia_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RKnee_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RKnee_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="RAnkle_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RAnkle_pitch">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RAnkle_pitch_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="RFoot_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="RAnkle_roll">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="RAnkle_roll_motor">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>


  <!--Add Gazebo Controller
  After the last transmission tag, add a gazebo controller tag as given below
  <gazebo>
  <plugin name="control"
  filename="libgazebo_ros_control.so">
  <robotNamespace>/</robotNamespace>
  </plugin>
  </gazebo>
  -->
  <gazebo>
    <plugin name="control"
    filename="libgazebo_ros_control.so">
    <robotNamespace>/</robotNamespace>
    </plugin>
  </gazebo>


   <!-- Add Self Collision tags.
  Similar to transmission tags, create the self-collision tags for each movable link in URDF. 
  <gazebo reference="link_n">
  <selfCollide>true</selfCollide>
  </gazebo>

  Note: Change “link_n” with the name of the link.

  -->

  <gazebo reference="LLeg">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="LHip">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="LFemur">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="LTibia">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="LAnkle">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="LFoot">
    <selfCollide>true</selfCollide>
  </gazebo>


  <gazebo reference="RLeg">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="RHip">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="RFemur">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="RTibia">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="RAnkle">
    <selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="RFoot">
    <selfCollide>true</selfCollide>
  </gazebo>

  


</robot>